controllers/volumehandler.go

```diff
@@ -261,6 +261,25 @@ func (h *destinationVolumeHandler) PreserveImage(l logr.Logger) (bool, error) {
 	return false, fmt.Errorf("unsupported copyMethod: %v -- must be None or Snapshot", h.Options.CopyMethod)
 }
 
+// PreserveImage implements the methods for preserving a PiT copy of the
+// replicated data.
+func (h *destinationVolumeHandler) PreserveRclone(l logr.Logger) (bool, error) {
```

Comment:
Why do we need a separate Preserve for rclone?
I see some difference in the steps for Snapshot, but I'm not following why it's necessary.

Reply:
I am encountering something weird here. Adding `h.removeSnapshotAnnotation` in `preserveImage` deletes the `volumesnapshot` before they are used by the controller to create `pvcs`
So the destination `pvc` is left in the pending state. Until then it is a fix.

Reply:
Interesting. I think you need to at least have the ordering as create => cleanup => record. I think putting record before cleanup may leave old snaps sitting around. latestImage is the only place where the previous snap name exists, and when you call record, it overwrites that with the new name.

Reply:
@JohnStrunk Looks like this is the issue. By the time we extract the latest snapshot image from `ReplicationDestination` and update `mysql-pvc.yaml`,  the current job completes, and the next reconciliation updates the latest image in `ReplicationDestination`  and deletes that volumesnapshot. Therefore the `pvc/mysql-pv-claim` is in pending state with the warning
```
 Warning  ProvisioningFailed    1s (x5 over 14s)  hostpath.csi.k8s.io_csi-hostpath-provisioner-0_89bbc19c-56d9-4016-b666-6aff5e9e6199  failed to provision volume with StorageClass "csi-hostpath-sc": error getting handle for DataSource Type VolumeSnapshot by Name scribe-dest-database-destination-20210113123131: error getting snapshot scribe-dest-database-destination-20210113123131 from api server: volumesnapshots.snapshot.storage.k8s.io "scribe-dest-database-destination-20210113123131" not found
```
Logged this in issue https://github.com/backube/scribe/issues/79